src/queries.rs

```diff
@@ -701,11 +698,25 @@ impl<'a> VintedWrapper<'a> {
 
         let json: Response = client.get(url).send().await?;
 
+        debug!("{:?}", &json);
+
         match json.status() {
             StatusCode::OK => {
-                let items: Items = json.json().await?;
-                Ok(items)
+                // First, get the response body as text to enable debugging if deserialization fails
+                let raw_json = json.text().await?;
```

Comment:
You cannot directly Deserialize as json ?

I understand the purpose you do this at first , in order to debug what was happening but now you should try to Deserialize as json directly.

Reply:
I'd like to leave it as it is so we can have better debug information while testing.

**However**, I think we should include it in `VintedWrapperError::SerdeError` as a param. I tried to do so and couldn't manage to implement it without breaking the previous code. I'd need ur help with that.

Performance-wise it doesn't have a big impact in my opinion:

### Using the `json.json()` method

```rust
    #[cfg(feature = "json")]
    #[cfg_attr(docsrs, doc(cfg(feature = "json")))]
    pub async fn json<T: DeserializeOwned>(self) -> crate::Result<T> {
        let full = self.bytes().await?;
        serde_json::from_slice(&full).map_err(crate::error::decode)
    }
```

```rust
pub fn from_slice<'a, T>(v: &'a [u8]) -> Result<T>
where
    T: de::Deserialize<'a>,
{
    from_trait(read::SliceRead::new(v))
}
```

### Storing Raw + Deserialize it

1. `json.text()`

```rust
            let full = self.bytes().await?;
           // These are the 2 extra steps it's taking
            let text = String::from_utf8_lossy(&full); //Convert the bytes into &String 
            Ok(text.into_owned()) //Clone it to a String
```

2. `from_str`

```rust
fn from_trait<'de, R, T>(read: R) -> Result<T>
where
    R: Read<'de>,
    T: de::Deserialize<'de>,
{
    let mut de = Deserializer::new(read);
    let value = tri!(de::Deserialize::deserialize(&mut de));

    // Make sure the whole stream has been consumed.
    tri!(de.end());
    Ok(value)
}
```

Reply:
[make Json transparent in the error](https://github.com/ThalosES/vinted-rs/pull/118/commits/d45f800f09339bacc15584715d32633c0024ec49)